java/src/jmri/jmrit/beantable/BeanTableDataModel.java

```diff
@@ -860,20 +860,31 @@ protected void showPopup(JmriMouseEvent e) {
         menuItem.addActionListener((ActionEvent e1) -> editComment(rowindex, 0));
         popupMenu.add(menuItem);
 
+        menuItem = new JMenuItem(Bundle.getMessage("CopySystemName"));
+        menuItem.addActionListener((ActionEvent e1) -> copySystemName(rowindex, 0));
+        popupMenu.add(menuItem);
+
         menuItem = new JMenuItem(Bundle.getMessage("ButtonDelete"));
         menuItem.addActionListener((ActionEvent e1) -> deleteBean(rowindex, 0));
         popupMenu.add(menuItem);
 
         popupMenu.show(e.getComponent(), e.getX(), e.getY());
     }
 
-    public void copyName(int row, int column) {
+    public void copyUserName(int row, int column) {
         T nBean = getBySystemName(sysNameList.get(row));
         Clipboard clipboard = Toolkit.getDefaultToolkit().getSystemClipboard();
         StringSelection name = new StringSelection(nBean.getUserName());
         clipboard.setContents(name, null);
     }
 
+    public void copySystemName(int row, int column) {
+        T nBean = getBySystemName(sysNameList.get(row));
+        Clipboard clipboard = Toolkit.getDefaultToolkit().getSystemClipboard();
+        StringSelection name = new StringSelection(nBean.getSystemName());
+        clipboard.setContents(name, null);
```

Comment:
No need to change, but this code looks more complicated than I think it needs to be.
If I understand it correctly, the system name is in sysNameList.get(row). So I don't think you need to look up the bean to get the system name.